docs/software-platforms/docker/docker-images.md

```diff
@@ -0,0 +1,114 @@
+The easiest way to do this is likely to be to use the same tool (`renv`, `conda`, etc.) that you used to define the software environment for your analysis module.
+If necessary, you can also include statements to install other software using the system package manager (e.g., `apt-get` for Debian-based images).
+
+## R-based images
+
+If you are working with R, you will likely want to use one of the [`rocker/r-ver` images](https://hub.docker.com/r/rocker/r-ver/tags) or a[`bioconductor/r-ver` image](https://hub.docker.com/r/bioconductor/r-ver/tags) images, which are a set of Docker images that provide complete R environments, tied to a specific version of R and/or Bioconductor.
+The Bioconductor images are especially handy as they include system software dependencies for Bioconductor packages that may not be installed in the `rocker` images, and support binary installation of Bioconductor packages, which can dramatically speed up build times.
+
+You will want to include the following steps in your Dockerfile:
+
+- Install `renv` (and `remotes` if you have any GitHub-based packages in your environment).
+- Copy the `renv.lock` file from the host environment to the image.
+- Use `renv::restore()` to reproduce the R environment from the `renv.lock` file.
+
+
+A simple `Dockerfile` for an R-based analysis module that uses `renv` for its environment might look like this:
+
+```Dockerfile
+# Base image on the Bioconductor 3.18 image
+FROM bioconductor/r-ver:3.18
+
+# Install renv to enable later package installation
+RUN Rscript -e "install.packages('renv')"
+
+# Disable the renv cache to install packages directly into the R library
+ENV RENV_CONFIG_CACHE_ENABLED FALSE
+
+# Copy the renv.lock file from the host environment to the image
+COPY renv.lock renv.lock
+
+# restore from renv.lock file and clean up to reduce image size
+RUN Rscript -e 'renv::restore()' && \
+  rm -rf ~/.cache/R/renv && \
+  rm -rf /tmp/downloaded_packages && \
+  rm -rf /tmp/Rtmp*
+```
+
+
+## Conda-based images
+
+If you are working with conda, you will likely want to use one of the [`continuumio/miniconda3` images](https://hub.docker.com/r/continuumio/miniconda3/tags).
+Since the conda environment files should define the version of Python itself, you should not need to worry too much about the specific version of the base image, but you should still use a versioned image for reproducibility.
+
+In the Dockerfile, you will want include the following steps:
+
+- Install `conda-lock`.
+- Copy the `conda-lock.yml` file from the host environment to the image.
+- Use `conda-lock install` to create an environment from the `conda-lock.yml` file.
+- Make sure the environment is activated when the container is launched by modifying the `.bashrc` file and setting the `ENTRYPOINT` to `bash -l -c`.
+
+A simple `Dockerfile` for a conda-based analysis module that uses `conda-lock` for its environment might look like this:
```

Comment:
```suggestion
A simple `Dockerfile` for a conda-based analysis module that uses `conda-lock` for its environment might begin like this:
```

Reply:
This is the whole thing though? (Aside from labels, as noted)

Reply:
Ah yeah, I suppose it is. I was thinking about what if more dependencies are needed, but they'd be handled with the conda lines!